#3 create a centralized logging module

TysonBeck opened the pull request.

TysonBeck committed:
#3 created new branch from master and adding in central logging files.

TysonBeck committed:
#3 Starting to factor centralized logging and agent into a standalone module

TysonBeck committed:
#3 Adding common variables

TysonBeck committed:
#3 Refactoring submodules

TysonBeck committed:
#3 Resolved output slap warnings.

TysonBeck committed:
Merge branch 'master' into issue/3-Create-a-centralized-logging-module

TysonBeck committed:
Merge branch 'master' into issue/3-Create-a-centralized-logging-module

TysonBeck committed:
#3 Added master/child deploy logic for combined module
Refactored module to new schema
Resolved output required splat error

jeremyodle commented on aws/central-logging/central-logging.tf:
```diff
@@ -0,0 +1,118 @@
+}
+
+variable "namespace" {
+  description = "Namespace (e.g. `cp` or `cloudposse`)"
```
Replace with "Namespace (e.g. `grv` or `gravicore`)"

jeremyodle commented on aws/central-logging/module.tf:
```diff
@@ -0,0 +1,59 @@
+  region  = "${var.aws_region}"
+
+  assume_role {
+    role_arn = "arn:aws:iam::${var.account_id}:role/grv_deploy_svc"
```
Utilize variable for role name.

jeremyodle commented on aws/central-logging/module.tf:
```diff
@@ -0,0 +1,59 @@
+  region  = "${var.aws_region}"
+
+  assume_role {
+    role_arn = "arn:aws:iam::${var.master_account_id}:role/grv_deploy_svc"
```
Utilize variable for role name.

jeremyodle commented on aws/central-logging/central-logging.tf:
```diff
@@ -0,0 +1,118 @@
+}
+
+variable "stage" {
+  description = "Stage (e.g. `prod`, `dev`, `staging`)"
```
Replace with "Stage (e.g. `dev`, `uat`, `prd`)"

jeremyodle commented on aws/central-logging/agent/module.tf:
```diff
@@ -0,0 +1,58 @@
+  region  = "${var.aws_region}"
+
+  assume_role {
+    role_arn = "arn:aws:iam::${var.master_account_id}:role/grv_deploy_svc"
```
Utilize variable for role_arn role name.

jeremyodle commented on aws/central-logging/agent/module.tf:
```diff
@@ -0,0 +1,58 @@
+  region  = "${var.aws_region}"
+
+  assume_role {
+    role_arn = "arn:aws:iam::${var.account_id}:role/grv_deploy_svc"
```
Utilize variable for role_arn role name.

jeremyodle commented on aws/central-logging/agent/central-logging-agent.tf:
```diff
@@ -0,0 +1,142 @@
+}
+
+variable "stage" {
+  description = "Stage (e.g. `prod`, `dev`, `staging`)"
```
Replace with "Stage (e.g. `dev`, `uat`, `prd`)"

jeremyodle commented on aws/central-logging/agent/central-logging-agent.tf:
```diff
@@ -0,0 +1,142 @@
+# ----------------------------------------------------------------------------------------------------------------------
+
+variable "namespace" {
+  description = "Namespace (e.g. `cp` or `cloudposse`)"
```
Replace with "Namespace (e.g. `grv` or `gravicore`)"

jeremyodle reviewed (changes requested):
- Update AWS provider assume role to utilize a variable for the role name
- Update variable descriptions
- Remove AWS provider references

TysonBeck committed:
#3 Modified variable description to current naming conventions
Variablelized role assumption
Removed redundant assume role in sub modules to resolve deploy errors

TysonBeck committed:
#3 Modified output to remove error due to output of map

jeremyodle reviewed (changes requested):
Move  `master_account_assume_role_name` into `module.tf`.
Example: https://github.com/gravicore/terraform-gravicore-modules/blob/master/aws/spoke-vpc/module.tf#L86

TysonBeck committed:
#3 Moved `master_account_assume_role_name` variable into module.tf files

jeremyodle reviewed (changes requested):
Missing README update for module.

TysonBeck committed:
#3 Updated readme for centralized logging module

jeremyodle commented on README.md:
```diff
@@ -16,6 +16,7 @@ This is a collection of reusable Terraform modules for Gravicore's cloud automat
 | [Aviatrix](aws/aviatrix)                     | A collection of modules for deploying [Aviatrix](https://www.aviatrix.com)'s global transit hub solution.                                                                                          |
 | [Instance Scheduler](aws/instance-scheduler) | A module enabling the configuration of custom start and stop schedules for Amazon EC2 and RDS instances. [Read More](https://aws.amazon.com/answers/infrastructure-management/instance-scheduler/) |
 | [RDS](aws/rds)                               | A collection of modules for deploying Relational Database Service (RDS) resources.                                                                                                                 |
+| [Central Logging](aws/central-logging)       | A collection of modules for deploying resources for Centralized logging.                                                                                                                           |
```
"A collection of modules for deploying resources for centralized logging."

TysonBeck committed:
#3 Corrected grammar in README
